tool.gpt

```diff
@@ -1,21 +1,16 @@
-tools: service, port, get-operating-system, open-file, open, start-file, start
-
-!!Important!! -  A file can be local or remote.
-
-Do all of the following in order and do not skip a step.
-1. Run the service.
-2. Get the port for the running service.
-3. Get the operating system.
-4. For the provided file, do one of the following two options:
-    - If it is remote or starts with github.com, do not edit it.
-    - If it is local, trim the directory from the file and remove the '.gpt' suffix.
-5. Do one of the following options.
-    - If the operating system is Windows:
-        - If a file or URL is provided, use the start file tool.
-        - If no file or URL is provided, use the start tool.
-    - If the operating system is not Windows:
-        - If a file or URL is provided, use the open-file tool.
-        - If a file or URL is not provided, use the open tool.
+context: github.com/gptscript-ai/context/os
+tools: service, port, open-file-nix, open-nix, open-file-windows, open-windows
+params: file: (optional) A reference to a tool file. Can refer to either a local or remote file.
+
+Never run steps in parallel.
+
+Perform the following steps in the order they appear:
+1. Run the service
+2. Get the port for the running service
+3. If no {file} was provided AND the local operating system is windows, use the open-windows tool then skip the remaining steps
+4. If no {file} was provided use the open-nix tool then skip the remaining steps
+5. If the operating system is windows, use the open-file-windows tool then skip the remaining steps
+6. Use the open-file-nix tool
```

Comment:
I think there's probably a more clever way to select the right tools based on OS via contexts, but ATM this was the most straightforward way I could come up with.

Reply:
I'm a little surprised that the LLM does the right thing and doesn't try to do _something_ for each listed step.

Reply:
I did some basic testing to make sure it works for remote and local paths, but I didn't try every combination here (e.g. on windows). I'll check on windows and take a look at the events to see exactly what the LLM does here.

Reply:
checked and remote URLs are working properly on windows